src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/Jira.kt

```diff
@@ -1,41 +1,113 @@
 import com.google.common.util.concurrent.ThreadFactoryBuilder
 import org.apache.logging.log4j.LogManager
 import org.apache.logging.log4j.Logger
 import java.net.URI
 import java.util.concurrent.Executors
 
-class Jira(
+/**
+ * @param extraResultsSources source of results/diagnostics of: reverse proxy, Crowd, LDAP, DVCS, DB, Jira plugins
+ * or anything that can be integrated with Jira as part of web application provisioning
+ */
+class Jira private constructor(
     private val nodes: List<StartedNode>,
     val jiraHome: RemoteLocation,
     val database: RemoteLocation?,
     val address: URI,
-    val jmxClients: List<JmxClient> = emptyList()
+    val jmxClients: List<JmxClient>,
+    private val extraResultsSources: List<MeasurementSource>
 ) : MeasurementSource {
     private val logger: Logger = LogManager.getLogger(this::class.java)
 
+    @Deprecated("Use Jira.Builder instead.")
+    constructor(
+        nodes: List<StartedNode>,
+        jiraHome: RemoteLocation,
+        database: RemoteLocation?,
+        address: URI,
+        jmxClients: List<JmxClient>
+    ) : this(
+        nodes = nodes,
+        jiraHome = jiraHome,
+        database = database,
+        address = address,
+        jmxClients = jmxClients,
+        extraResultsSources = emptyList()
+    )
+
+    @Deprecated("Use Jira.Builder instead.")
+    constructor(
+        nodes: List<StartedNode>,
+        jiraHome: RemoteLocation,
+        database: RemoteLocation?,
+        address: URI
+    ) : this(
+        nodes = nodes,
+        jiraHome = jiraHome,
+        database = database,
+        address = address,
+        jmxClients = emptyList(),
+        extraResultsSources = emptyList()
+    )
+
     override fun gatherResults() {
-        if (nodes.isEmpty()) {
-            logger.warn("No Jira nodes known to JPT, not downloading node results")
+        val firstNode = nodes.firstOrNull()
+        val measurementSources = extraResultsSources + nodes + listOfNotNull(firstNode?.let { AnalyticsLogsSource(it) })
+        if (measurementSources.isEmpty()) {
+            logger.warn("No result sources known to Jira, can't download anything")
             return
         }
         val executor = Executors.newFixedThreadPool(
-            nodes.size.coerceAtMost(4),
+            measurementSources.size.coerceAtMost(4),
             ThreadFactoryBuilder()
                 .setNameFormat("results-gathering-thread-%d")
                 .build()
         )
-        nodes.map { executor.submitWithLogContext("gather $it") { it.gatherResults() } }
-            .forEach { it.get() }
-        time("gather analytics") { nodes.firstOrNull()?.gatherAnalyticLogs() }
+        SuppressingRunnable(
+            measurementSources.map { executor.submitWithLogContext("gather $it") { it.gatherResults() } }
+                .map { Runnable { it.get() } }
+        ).run()
+
         executor.shutdownNow()
     }
 
     override fun toString() = "Jira(address=$address)"
+
+    class Builder(
```

Comment:
I thought that I should probably use the same required constructor params as in https://github.com/atlassian/aws-infrastructure/pull/162/files#diff-100d5a9943e653f7807abf68bf52b9fb2d04cb317126a4b1b4fec5f3d014f03d, however I really wanted to make optional as many of them as possible, so that the consumer could build `Jira` easier, e.g. see https://github.com/atlassian/aws-infrastructure/pull/163/files#diff-bd4b59a5570c8b384c9626a7e8ea1f86f6d1cab564e86aa1f77667a383b731ff

Reply:
We paired on this with @ewefie and we decided that we don't want to merge it without @dagguh's review.

@dagguh this is the main uncertainity we have in this PR. Please share your point of view.